### PR TITLE
[docs] New K8s-Consul deployment guide

### PR DESCRIPTION
**Pull request by kaitlincarter-hc**
This guide covers deploying Consul with some production recommendations and includes a complete helm chart example.

### CODE REVIEW

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+can reference these secrets in the customized values file. 
+
+If you have purchased Enterprise Consul, the enterprise license file should be
+used with the official image,  `hashicorp/consul-enterprise:1.5.0-ent`.  Enable
+[encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption)
+for gossip traffic. 
+
+
+~> Note, depending on your environment, the previous secrets may not be
+necessary.  ## Configure Helm Chart
```
Comment (kaitlincarter-hc) [inline website/source/docs/guides/kubernetes-production-deploy.md]:
```suggestion
necessary.  

## Configure Helm Chart 
```

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+functionality including Connect and the Consul UI. 
+
+Now you can interact with your Consul cluster through the UI or CLI. 
+
+If you exposed the UI using a load balancer it will be available at the
+`LoadBalancer Ingress` IP address and `Port` that is output from the following
+command.
+
+```sh 
+$ kubectl describe services consul-server
```
Comment (judithpatudith) [inline website/source/docs/guides/kubernetes-production-deploy.md]: I think`consul-server` here will have random animal names in front of it if we don't instruct people to name their deployment. Wasn't sure how to indicate this in the command.

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  domain: consul
+  datacenter: primarydc
+  # Bootstrap ACLs within Consul. This is highly recommended.
+  bootstrapACLs: true
+  # Gossip encryption
+  gossipEncryption: |
+    secretName: "encrypt-key"
+    secretKey: "key
+# Configure your Consul servers in this section.
+server:
```
Comment (adilyse) [inline website/source/docs/guides/kubernetes-production-deploy.md]: You mention earlier that we're enabling connect on the servers, but the `server:connect:true` setting is missing from this example.

File: website/source/docs/guides/kubernetes-production-deploy.md (hunk 2 above)
Comment (adilyse) [reply website/source/docs/guides/kubernetes-production-deploy.md]: Yep, like my `reeling-quoll-consul-server` I'm rocking right now :).

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+Once you have a service account, you will also need to add a permission to
+deploy the helm chart. This is done with the `clusterrolebinding` method. 
+
+```sh
+$ kubectl create clusterrolebinding kubernetes-dashboard -n kube-system --clusterrole=cluster-admin --serviceaccount=kube-system:kubernetes-dashboard
+```
+
+Finally, you may need to create Kubernetes secrets to store Consul data. You
+can reference these secrets in the customized values file.
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: I think so!

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  resources: |
+    requests:
+      memory: "32Gi"
+      cpu: "4"
+      disk: "50Gi"
+    limits:
+      memory: "32Gi"
+      cpu: "4"
+      disk: "50Gi"
+  # If using Enterprise, provide your license here
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: Good catch!

File: website/source/docs/guides/kubernetes-production-deploy.md (hunk 2 above)
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: I added a note!

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+```sh
+$ kubectl create clusterrolebinding kubernetes-dashboard -n kube-system --clusterrole=cluster-admin --serviceaccount=kube-system:kubernetes-dashboard
+```
+
+Finally, you may need to create Kubernetes secrets to store Consul data. You
+can reference these secrets in the customized values file. 
+
+If you have purchased Enterprise Consul, the enterprise license file should be
+used with the official image,  `hashicorp/consul-enterprise:1.5.0-ent`.  Enable
+[encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption)
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: I updated the sentence to be more descriptive.

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+parameters](https://www.consul.io/docs/platform/k8s/helm.html#v-global).
+
+### Consul UI
+
+To enable the Consul web UI update the `ui` section to your values file and set
+`enabled` to `true`. 
+
+Note, you can also set up a [loadbalancer
+resource](https://github.com/hashicorp/demo-consul-101/tree/master/k8s#implement-load-balancer)
+or other service type in Kubernetes to make it easier to access the UI.  ###
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: Fixed!

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+reference the Kubernetes secret in the `enterpriseLicense` parameter.
+
+Read the Consul Helm chart documentation to review all the [server
+parameters](https://www.consul.io/docs/platform/k8s/helm.html#v-server)
+
+### Consul Clients
+
+A Consul client is deployed on every Kubernetes node, so you do not need to
+specify the number of clients for your deployments. You will need to specify
+resources and enable gRPC. The resources in the example below should be
```
[the remark this answers is not in the document]
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: Fixed!

File: website/source/docs/guides/kubernetes-production-deploy.md (hunk 3 above)
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: Updated!

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+
+#### Affinity
+
+To ensure the Consul servers are placed on different Kubernetes nodes, you will
+need to configure affinity. Otherwise, the failure of one Kubernetes node could
+cause the loss of multiple Consul servers, and result in quorum loss. By
+default, the example `values.yaml` has affinity configured correctly.  
+
+####
+Enterprise License
```
Comment (adilyse) [inline website/source/docs/guides/kubernetes-production-deploy.md]: This should be on the previous line with the ####, right?

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+
+Even though you enabled Consul server communication over Connect in the server section, you will also
+need to enable `connectInject` by setting `enabled` to `true`. In the
+`connectInject` section you will also configure security features. Enabling the
+`default` parameter will allow the injector to automatically inject the Connect
+sidecar into all pods. If you would prefer to manually annotate which pods to inject, you
+can set this to false. Setting the 'aclBindingRuleSelector` parameter to
+`serviceaccount.name!=default` ensures that new services do not all receive the
+same token if you are only using a default service account. This setting is
+only necessary if you have enabled ACLs in the server section.
```
Comment (adilyse) [inline website/source/docs/guides/kubernetes-production-deploy.md]:
```suggestion
only necessary if you have enabled ACLs in the global section.
```

File: website/source/docs/guides/kubernetes-production-deploy.md (hunk 9 above)
Comment (kaitlincarter-hc) [reply website/source/docs/guides/kubernetes-production-deploy.md]: Correct!

Comment (kaitlincarter-hc) [inline website/source/docs/guides/kubernetes-production-deploy.md]:
```suggestion
#### Enterprise License
```

File: website/source/docs/guides/kubernetes-production-deploy.md [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+example at the end of the guide, the resources are set for a large environment. 
+
+#### Affinity
+
+To ensure the Consul servers are placed on different Kubernetes nodes, you will
+need to configure affinity. Otherwise, the failure of one Kubernetes node could
+cause the loss of multiple Consul servers, and result in quorum loss. By
+default, the example `values.yaml` has affinity configured correctly.  
+
+####
```
Comment (kaitlincarter-hc) [inline website/source/docs/guides/kubernetes-production-deploy.md]:
```suggestion
```

**Pull request merged**